corehq/apps/users/dbaccessors/all_commcare_users.py

```diff
@@ -182,6 +188,30 @@ def get_deleted_user_by_username(cls, username):
     return cls.wrap_correctly(result['doc']) if result else None
 
 
+def user_exists(username):
+    """
+    :param username:
+    :return: namedtuple(exists:bool, is_deleted:bool)
+    """
+    result = CommCareUser.get_db().view(
+        'users/by_username',
+        key=username,
+        include_docs=False,
+        reduce=False,
+    )
+    if result:
+        return UserExists(True, False)
```

Comment:
When someone adds a new user with the same username that has been deleted, it would have two couch docs with the same name but only one django user?

This returns `False` without checking if that username is present for a deleted user or not. Guess thats because if a user exists with that username checking for a deleted user is irrelevant. But in case someone adds a username which was deleted earlier, this should return True for both as well? Or that wont ever happen?

Reply:
This return value is correct. If both are True then the user will just get a warning. Having (True, False) indicates that there is a 'live' user with that username which will block any new users being created.